### PR TITLE
ovl en sob1 OK and Documented

### PR DESCRIPTION
**Pull request by hensldm**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Shopkeeper Actors (3/4)

### CODE REVIEW

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 1]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+
+    this->stickAnimTween = stickAnimTween;
+
+    this->stickLeftPrompt.arrowColorR = COL_CHAN_MIX(255, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorG = COL_CHAN_MIX(new_var2, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorB = COL_CHAN_MIX(0, -100, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorA = COL_CHAN_MIX(200, 50.0f, arrowAnimTween);
+
+    this->stickRightPrompt.arrowColorR = (new_var2 - ((s32)new_var3)) & 0xFF;
+    this->stickRightPrompt.arrowColorG = (255 - ((s32)new_var3)) & 0xFF;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1262]: Could those two `& 0xFF` be `(u8)` casts?
```suggestion
    this->stickRightPrompt.arrowColorR = (u8)(new_var2 - ((s32)new_var3));
    this->stickRightPrompt.arrowColorG = (u8)(255 - ((s32)new_var3));
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 2]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+    this->lookFowardCutscene = this->actor.cutscene;
+    this->lookToShelfCutscene = ActorCutscene_GetAdditionalCutscene(this->lookFowardCutscene);
+    this->lookToShopkeeperCutscene = ActorCutscene_GetAdditionalCutscene(this->lookToShelfCutscene);
+}
+
+void EnSob1_WaitForBlink(EnSob1* this) {
+    s16 decr;
+
+    decr = this->blinkTimer - 1;
+    if (decr != 0) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1306]:
```suggestion
    s16 decr = this->blinkTimer - 1;

    if (decr != 0) {
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 3]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+        this->blinkFunc = EnSob1_WaitForBlink;
+    } else {
+        this->eyeTextureIdx = eyeTextureIdxTemp;
+        this->blinkTimer = 1;
+    }
+}
+
+void EnSob1_Obj3ToSeg6(EnSob1* this, GlobalContext* globalCtx) {
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->objIndicies[2]].segment);
+}
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1334]: I feel like this function could have a better name. What about `_ChangeObject` or similar?

File: src/overlays/actors/ovl_En_In/z_en_in.h [hunk 4]
```diff
@@ -28,7 +28,7 @@ typedef struct EnIn {
     /* 0x238 */ char unk238[0x4];
     /* 0x23C */ u8 unk23C;
     /* 0x23D */ u8 unk23D;
-    /* 0x240 */ u8 *unk240;
+    /* 0x240 */ Path *path;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_In/z_en_in.h:31]: Our style is `Type* varname`; the format script doesn't do headers which is why it didn't catch this, presumably.

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 5]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+            this->items[i] = NULL;
+        } else {
+            this->items[i] =
+                (EnGirlA*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_GIRLA, shopItem->spawnPos.x,
+                                      shopItem->spawnPos.y, shopItem->spawnPos.z, 0, 0, 0, shopItem->shopItemId);
+        }
+    }
+}
+
+s32 EnSob1_GetObjIndicies(EnSob1* this, GlobalContext* globalCtx, s16* objIds) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:371]: `Indices` (Applies in many other places.)

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 6]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+}
+
+void EnSob1_EndWalk(EnSob1* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 distSq;
+    s16 sp2E;
+    s16 sp2C;
+
+    sp2E = this->skelAnime.animCurrentFrame / this->skelAnime.animPlaybackSpeed;
+    sp2C = SkelAnime_GetFrameCount(&D_06009120.common) / (s16)this->skelAnime.animPlaybackSpeed;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:736]:
```suggestion
    s16 sp2E = this->skelAnime.animCurrentFrame / this->skelAnime.animPlaybackSpeed;
    s16 sp2C = SkelAnime_GetFrameCount(&D_06009120.common) / (s16)this->skelAnime.animPlaybackSpeed;
    
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 7]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+}
+
+s32 EnSob1_OverrideLimbDrawZoraShopkeeper(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                          Actor* thisx) {
+    EnSob1* this = THIS;
+
+    if (limbIndex == 15) {
+        rot->x += this->headRot;
+    }
+    return 0;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1617]:
```suggestion
    return false;
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 8]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+}
+
+s32 EnSob1_OverrideLimbDrawBombShopkeeper(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                          Actor* thisx) {
+    EnSob1* this = THIS;
+
+    if (limbIndex == 15) {
+        SysMatrix_InsertXRotation_s(this->headRot, MTXMODE_APPLY);
+    }
+    return 0;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1627]:
```suggestion
    return false;
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 9]
```diff
@@ -8,7 +14,68 @@ void EnSob1_Init(Actor* thisx, GlobalContext* globalCtx);
...
+extern UNK_TYPE D_06005458;
+extern UNK_TYPE D_060050A0;
+extern UNK_TYPE D_060058A0;
+extern UNK_TYPE D_060060A0;
+extern UNK_TYPE D_06010438;
+extern UNK_TYPE D_06010C38;
+extern UNK_TYPE D_06011038;
+
+const char filname1[] = "../z_en_soB1.c"; // Unused
+const char filname2[] = "../z_en_soB1.c"; // Unused
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:71]: Typo?

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 10]
```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
...
+            break;
+        case BOMB_SHOP:
+            this->shopType = BOMB_SHOP;
+            break;
+        default:
+            Actor_MarkForDeath(&this->actor);
+            return;
+    }
+
+    objIds = sObjectIds[this->shopType];
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:415]: I think the uses of this variable would be easier to follow if there was at least an enum for the entry indices (something like what the three objects are actually used for).

**Review by EllipticEllipsis — COMMENTED** [review]
Nice once again. A few of my comments on EnOssan also apply here; again, most are fairly minor, although one's a pretty big find/replace job!

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c (hunk 1 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1262]: That doesn't match

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c (hunk 9 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:71]: Yeah, though I'm going to change the implementation to use `(void)"../z_en_soB1.c"` like `en_ossan`

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c (hunk 10 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:412]: That is a good idea, though I don't know enough about the objects yet to that. I've added a todo to add that enum when the objects are better understood.

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 11]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+    Vec3f selectedItemPosition = sSelectedItemPositions[this->shopType];
+    u8 i = this->cursorIdx;
+    EnGirlA* item;
+    ShopItem* shopItem;
+    Vec3f worldPos;
+
+    shopItem = &sShops[this->shopType][i];
+    item = this->items[i];
+
+    VEC3F_LERPIMPDST(&worldPos, &shopItem->spawnPos, &selectedItemPosition, this->shopItemSelectedTween);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1129]:
```suggestion
    ShopItem* shopItem = &sShops[this->shopType][i];
    Vec3f worldPos;

    item = this->items[i];

    VEC3F_LERPIMPDST(&worldPos, &shopItem->spawnPos, &selectedItemPosition, this->shopItemSelectedTween);
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 12]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+            }
+        }
+    }
+}
+
+void EnSob1_UpdateCursorAnim(EnSob1* this) {
+    f32 t;
+
+    t = this->cursorAnimTween;
+    if (this->cursorAnimState == 0) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1193]:
```suggestion
    f32 t = this->cursorAnimTween;

    if (this->cursorAnimState == 0) {
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 13]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+
+    this->stickAnimTween = stickAnimTween;
+
+    this->stickLeftPrompt.arrowColorR = COL_CHAN_MIX(255, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorG = COL_CHAN_MIX(maxColor, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorB = COL_CHAN_MIX(0, -100, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorA = COL_CHAN_MIX(200, 50.0f, arrowAnimTween);
+
+    this->stickRightPrompt.arrowColorR = (maxColor - ((s32)tmp)) & 0xFF;
+    this->stickRightPrompt.arrowColorG = (255 - ((s32)tmp)) & 0xFF;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1257]:
```suggestion
    this->stickRightPrompt.arrowColorR = (u8)(maxColor - ((s32)tmp));
    this->stickRightPrompt.arrowColorG = (u8)(255 - ((s32)tmp));
```
Would this still work?

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 14]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+        points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+        points = &points[pointIdx];
+        diffX = points->x - pos->x;
+        diffZ = points->z - pos->z;
+    } else {
+        diffX = 0.0f;
+        diffZ = 0.0f;
+    }
+    *distSq = SQ(diffX) + SQ(diffZ);
+    return Math_Acot2F(diffZ, diffX) * (0x8000 / M_PI);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1288]:
```suggestion
    return RADF_TO_BINANG(Math_Acot2F(diffZ, diffX));
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 15]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+        if ((this->shopType == BOMB_SHOP) && (gSaveContext.weekEventReg[0x21] & 8)) {
+            sShops[this->shopType][0].shopItemId = SI_BOMB_BAG_30_2;
+        }
+
+        this->cutsceneState = 0;
+        EnSob1_GetCutscenes(this);
+        this->cutscene = this->lookFowardCutscene;
+        ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 20.0f);
+        sInitFuncs[this->shopType](this, globalCtx);
+        this->actor.colChkInfo.mass = 0xFF;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1398]:
```suggestion
        this->actor.colChkInfo.mass = MASS_IMMOVABLE;
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 16]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+
+    return dList;
+}
+
+void EnSob1_DrawZoraShopkeeper(Actor* thisx, GlobalContext* globalCtx) {
+    static TexturePtr sZoraShopkeeperEyeTextures[] = { &D_060050A0, &D_060058A0, &D_060060A0 };
+    
+    EnSob1* this = THIS;
+    s32 pad;
+    s32 i;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1651]:
```suggestion
    static TexturePtr sZoraShopkeeperEyeTextures[] = { &D_060050A0, &D_060058A0, &D_060060A0 };
    EnSob1* this = THIS;
    s32 pad;
    s32 i;
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 17]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gDPSetEnvColor(POLY_OPA_DISP++, 0x00, 0x00, 0x00, 0xFF);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1655]:
```suggestion
    gDPSetEnvColor(POLY_OPA_DISP++, 0, 0, 0, 255);
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 18]
```diff
@@ -21,164 +85,1636 @@ const ActorInit En_Sob1_InitVars = {
...
+    EnSob1_DrawStickDirectionPrompt(globalCtx, this);
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+}
+
+void EnSob1_DrawGoronShopkeeper(Actor* thisx, GlobalContext* globalCtx) {
+    static TexturePtr sGoronShopkeeperEyeTextures[] = { &D_06010438, &D_06010C38, &D_06011038 };
+    
+    EnSob1* this = THIS;
+    s32 pad;
+    s32 i;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1675]:
```suggestion
    static TexturePtr sGoronShopkeeperEyeTextures[] = { &D_06010438, &D_06010C38, &D_06011038 };
    EnSob1* this = THIS;
    s32 pad;
    s32 i;
```

**Review by engineer124 — COMMENTED** [review]
Great work! Here's my small feedback

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c (hunk 13 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:1257]: In its current state that does not help. I hope to eventually get the function into a state that these can also use the COL_CHAN_MIX, assuming that's possible.

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 19]
```diff
@@ -21,164 +86,1626 @@ const ActorInit En_Sob1_InitVars = {
...
+            ActorCutscene_SetIntentToPlay(this->cutscene);
+        }
+    }
+}
+
+void EnSob1_EndWalk(EnSob1* this, GlobalContext* globalCtx) {
+    s32 pad;
+    f32 distSq;
+    s16 sp2E = this->skelAnime.animCurrentFrame / this->skelAnime.animPlaybackSpeed;
+    s16 sp2C = SkelAnime_GetFrameCount(&D_06009120.common) / (s16)this->skelAnime.animPlaybackSpeed;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:725]:
```suggestion
    s16 animLastFrame = SkelAnime_GetFrameCount(&D_06009120.common) / (s16)this->skelAnime.animPlaybackSpeed;
```

File: src/overlays/actors/ovl_En_Sob1/z_en_sob1.c [hunk 20]
```diff
@@ -21,164 +86,1626 @@ const ActorInit En_Sob1_InitVars = {
...
+
+    Math_SmoothStepToS(
+        &this->actor.world.rot.y,
+        EnSob1_GetXZAngleAndDistanceSqToPoint(this->path, this->pathPointsIdx - 1, &this->actor.world.pos, &distSq), 4,
+        1000, 1);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    Math_ApproachF(&this->actor.speedXZ, 0.5f, 0.2f, 1.0f);
+    if (distSq < 12.0f) {
+        this->actor.speedXZ = 0.0f;
+        if (sp2C == sp2E) {
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Sob1/z_en_sob1.c:735]:
```suggestion
        if (animLastFrame == animCurrentFrame) {
```

**Review by Kenix3 — CHANGES_REQUESTED** [review]
Great work!
One final thing left needed to finish documentation.

**Review by Kenix3 — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

**Pull request merged**